user_token.go

```diff
@@ -0,0 +1,135 @@
+package tfe
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/url"
+	"time"
+)
+
+// Compile-time proof of interface implementation.
+var _ UserTokens = (*userTokens)(nil)
+
+// UserTokens describes all the user token related methods that the
+// Terraform Cloud/Enterprise API supports.
+//
+// TFE API docs:
+// https://www.terraform.io/docs/enterprise/api/user-tokens.html
+type UserTokens interface {
+	// List all the tokens of the given user ID.
+	List(ctx context.Context, userID string) (*UserTokenList, error)
+
+	// Generate a new user token
+	Generate(ctx context.Context, userID string, options UserTokenGenerateOptions) (*UserToken, error)
+
+	// Read a user token by its ID.
+	Read(ctx context.Context, tokenID string) (*UserToken, error)
+
+	// Delete a user token by its ID.
+	Delete(ctx context.Context, tokenID string) error
+}
+
+// userTokens implements UserTokens.
+type userTokens struct {
+	client *Client
+}
+
+// UserTokenList is a list of tokens for the given user ID.
+type UserTokenList struct {
+	*Pagination
+	Items []*UserToken
+}
+
+// UserToken represents a Terraform Enterprise user token.
+type UserToken struct {
+	ID          string    `jsonapi:"primary,authentication-tokens"`
+	GeneratedAt time.Time `jsonapi:"attr,created-at,iso8601"`
```

Comment:
One last, tiny nit: Although the other tokens use `Generate()`, they still use a field name that matches the attribute: `CreatedAt`; we should do that here (again, just for consistency)

https://github.com/hashicorp/go-tfe/blob/19dc0b8eec47f6889541bd5b4ee6e4a81b48a908/team_token.go#L36-L42

https://github.com/hashicorp/go-tfe/blob/19dc0b8eec47f6889541bd5b4ee6e4a81b48a908/organization_token.go#L36-L42

Reply:
My mistake, a victim of `s/Create/Generate`. Fixed in a2e3be5